api/tests/test_auth.rs

```diff
@@ -0,0 +1,136 @@
+#[cfg(test)]
+mod tests {
+    use actix_web::http::StatusCode;
+    use actix_web::middleware::from_fn;
+    use actix_web::{App, test, web};
+    use boom_api::auth::{auth_middleware, get_default_auth};
+    use boom_api::conf::AppConfig;
+    use boom_api::db::get_default_db;
+    use boom_api::routes;
+    use mongodb::Database;
+
+    /// Test POST /auth
+    #[actix_rt::test]
+    async fn test_post_auth() {
+        let database: Database = get_default_db().await;
+        let auth_app_data = get_default_auth(&database).await.unwrap();
+        let app = test::init_service(
+            App::new()
+                .app_data(web::Data::new(database.clone()))
+                .app_data(web::Data::new(auth_app_data.clone()))
+                .service(routes::users::post_user)
+                .service(routes::auth::post_auth),
+        )
+        .await;
+
+        // Create a new user with a UUID username
+        let random_name = uuid::Uuid::new_v4().to_string();
+
+        let new_user = serde_json::json!({
+            "username": random_name,
+            "email":
+            format!("{}@example.com", random_name),
+            "password": "password123"
+        });
+
+        let req = test::TestRequest::post()
```

Comment:
Since we never delete this user, our local test databases are going to accumulate users each time we run the tests. Not the end of the world, but you could alternatively try to authenticate with the admin credentials in the config.